tools/autograd/derivatives.yaml

```diff
@@ -733,11 +733,11 @@
   self: value_selecting_reduction_backward(grad, dim, indices, self.sizes(), keepdim)
 
 - name: mul.Tensor(Tensor self, Tensor other) -> Tensor
-  self: grad * other
-  other: grad * self
+  self: mul_tensor_backward(grad, other, self.scalar_type())
+  other: mul_tensor_backward(grad, self, other.scalar_type())
 
 - name: mul.Scalar(Tensor self, Scalar other) -> Tensor
-  self: grad * other
+  self: mul_scalar_backward(grad, other, self.scalar_type())
```

Comment:
I think you can remove the `mul_scalar_backward` to reduce code duplication and just call `mul_tensor_backward(grad, at::scalar_to_tensor(other), self.scalar_type())`.

Reply:
good idea! thanks